src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -197,15 +211,32 @@ public IRecordingDescriptor startRecording(
                     if (fixedDuration != null) {
                         Long delay =
                                 Long.valueOf(fixedDuration.toString().replaceAll("[^0-9]", ""));
-
                         scheduleRecordingTasks(
                                 recordingName, delay, connectionDescriptor, archiveOnStop);
                     }
-
                     return desc;
                 });
     }
 
+    private boolean shouldRestartRecording(ReplacementPolicy replace) {
```

Comment:
Should this perhaps be like `shouldRestartRecording(ReplacementPolicy replace, IRecordingDescriptor descriptor)` so that the `descriptor.getState()` can also be checked?